lms/djangoapps/lms_xblock/mixin.py

```diff
@@ -150,11 +208,17 @@ def validate(self):
         validation = super(LmsBlockMixin, self).validate()
         has_invalid_user_partitions = False
         has_invalid_groups = False
+
         for user_partition_id, group_ids in self.group_access.iteritems():
             try:
                 user_partition = self._get_user_partition(user_partition_id)
             except NoSuchUserPartitionError:
                 has_invalid_user_partitions = True
+            except AttributeError:
+                try:
+                    user_partition = self.user_partitions[user_partition_id]
```

Comment:
Line 214 calls _get_user_partition which very occasionally would throw an AttributeError.  I'm not completely sure why that happened but it seemed to be an issue with the block's runtime.
The second try just tries to directly access the user_partitions attribute which would throw a KeyError if that user_partition does not exist in self.user_partitions.